src/ascii_string.rs

```diff
@@ -466,6 +466,20 @@ impl fmt::Debug for AsciiString {
     }
 }
 
+impl fmt::Write for AsciiString {
```

Comment:
I'll add some documentation that `fmt::Write` methods "[do] not support transmission of an error other than that an error occurred." (from `std::fmt::Error`). This is just to remind users of this API how to handle errors correctly.